Source/Modules/python.cxx

```diff
@@ -2060,120 +2119,34 @@ class PYTHON:public Language {
-
-    // Check if this is a number in any base.
-    long value = strtol(s, &end, 0);
-    (void) value;
-    if (end != s) {
-      if (errno == ERANGE) {
-	// There was an overflow, we could try representing the value as Python
-	// long integer literal, but for now don't bother with it.
-	fail = true;
-      } else {
-	if (*end != '\0') {
-	  // If there is a suffix after the number, we can safely ignore any
-	  // combination of "l" and "u", but not anything else (again, stuff like
-	  // "LL" could be handled, but we don't bother to do it currently).
-	  bool seen_long = false;
-	  for (char* p = end; *p != '\0'; ++p) {
-	    switch (*p) {
-	      case 'l':
-	      case 'L':
-		// Bail out on "LL".
-		if (seen_long) {
-		  fail = true;
-		  break;
-		}
-		seen_long = true;
-		break;
+    SwigType *resolved_type = SwigType_typedef_resolve_all(type);
 
-	      case 'u':
-	      case 'U':
-		break;
-
-	      default:
-		// Except that our suffix could actually be the fractional part of
-		// a floating point number, so we still have to check for this.
-		result = convertDoubleValue(v);
-	    }
-	  }
-	}
-
-	if (!fail) {
-	  // Allow integers as the default value for a bool parameter.
-	  resolved_type = SwigType_typedef_resolve_all(type);
-	  if (Cmp(resolved_type, "bool") == 0) {
-	    result = NewString(value ? "True" : "False");
-	  } else {
-	    // Deal with the values starting with 0 first as they can be octal or
-	    // hexadecimal numbers or even pointers.
-	    if (s[0] == '0') {
-	      if (Len(v) == 1) {
-		// This is just a lone 0, but it needs to be represented differently
-		// in Python depending on whether it's a zero or a null pointer.
-		if (SwigType_ispointer(resolved_type))
-		  result = NewString("None");
-		else
-		  result = v;
-	      } else if (s[1] == 'x' || s[1] == 'X') {
-		// This must have been a hex number, we can use it directly in Python,
-		// so nothing to do here.
-	      } else {
-		// This must have been an octal number. We will let python convert it
-		// to an int using base 8, since we would have to change its prefix
-		// to be "0o" in Python 3 only.
-		if (end - s > 1) {
-		  result = NewString("int('");
-		  String *octal_string = NewStringWithSize(s + 1, (int)(end - s - 1));
-		  Append(result, octal_string);
-		  Append(result, "', 8)");
-		  Delete(octal_string);
-		}
-	      }
-	    }
-
-	    // Avoid unnecessary string allocation in the common case when we don't
-	    // need to remove any suffix.
-	    if (!result)
-	      result = *end == '\0' ? v : NewStringWithSize(s, (int)(end - s));
-	  }
-	}
-      }
-    }
-
-    // Check if this is a floating point number (notice that it wasn't
-    // necessarily parsed as a long above, consider e.g. ".123").
-    if (!fail && !result) {
+    result = convertIntegerValue(v, resolved_type);
+    if (!result) {
       result = convertDoubleValue(v);
       if (!result) {
-	if (Strcmp(v, "true") == 0 || Strcmp(v, "TRUE") == 0)
```

Comment:
Yes, I fully agree. Actually, someone may define FALSE as -1 and use this as a default value for an int, so this could cause a real problem (and it will never do any good).

I'll add a commit to throw those tokens out.